src/Nix/Builtins.hs

```diff
@@ -250,7 +251,7 @@ nixPath = fmap nvList $ flip foldNixPath [] $ \p mn rest ->
 
 toString :: MonadNix e m => m (NValue m) -> m (NValue m)
 toString str =
-    str >>= normalForm >>= valueText False >>= toNix @(Text, DList Text)
+    str >>= normalForm >>= valueText False >>= toNix @NixString
```

Comment:
I really like using `NixString` to abstract this tuple, and now can't believe it hadn't been done until now. :)

Reply:
Yeah, this is vastly more understandable.